pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/config.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promtail
+
+import (
+	"fmt"
+	"time"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components"
+	"github.com/gardener/gardener/pkg/utils"
+	"gopkg.in/yaml.v3"
+	"k8s.io/utils/pointer"
+)
+
+const setActiveJournalFileScript = `#!/bin/bash
+PERSISTANT_JOURNAL_FILE=/var/log/journal
+TEMP_JOURNAL_FILE=/run/log/journal
+if [ ! -d "$PERSISTANT_JOURNAL_FILE" ] && [ -d "$TEMP_JOURNAL_FILE" ]; then
+	sed -i -e "s|$PERSISTANT_JOURNAL_FILE|$TEMP_JOURNAL_FILE|g" ` + PathPromtailConfig + `
+fi`
+
+type config struct {
+	Server        server        `yaml:"server"`
```

Comment:
why are all those fields exported? (same for other types below)

Reply:
I didn't pay any attention because the structure itself is not exposed. Sorry